File: veloce-backend/app/passwords.py
import bcrypt


def hash_password(password: str) -> str:
    pw = password[:72].encode("utf-8")
    return bcrypt.hashpw(pw, bcrypt.gensalt()).decode("ascii")


def verify_password(plain: str, hashed: str) -> bool:
    try:
        return bcrypt.checkpw(
            plain[:72].encode("utf-8"),
            hashed.encode("ascii"),
        )
    except (ValueError, TypeError):
        return False

File: veloce-frontend/src/lib/server/passwords.ts
import bcrypt from "bcrypt";

const BCRYPT_ROUNDS = 12;

export function hashPassword(password: string): string {
  const pw = Buffer.from(password.slice(0, 72), "utf8");
  return bcrypt.hashSync(pw, BCRYPT_ROUNDS);
}

export function verifyPassword(plain: string, hashed: string): boolean {
  try {
    const pw = Buffer.from(plain.slice(0, 72), "utf8");
    return bcrypt.compareSync(pw, hashed);
  } catch {
    return false;
  }
}
